File: frontend/src/types/category.ts
import type { BaseQuery } from "./baseQuery";

export interface Category {
  id: number;
  name: string;
  description: string;
  createdAt: string;
  updatedAt: string;
}

export interface CategoryRequest {
  name: string;
  description: string;
}

export interface CategoryQuery extends BaseQuery {
  search?: string;
  unpage?: boolean;
}

export interface CategoryError {
  name: string;
  description: string;
}

File: backend/internal/model/category.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Category struct {
	ID          uint   `gorm:"primaryKey"`
	Name        string `gorm:"type:varchar(255);not null;uniqueIndex:idx_name_deleted_at;index"`
	Description string `gorm:"not null;type:text"`
	CreatedAt   time.Time
	UpdatedAt   time.Time
	DeletedAt   gorm.DeletedAt `gorm:"index;uniqueIndex:idx_name_deleted_at"`
}
